python/hail/typ.py

```diff
@@ -91,16 +91,17 @@ def _typecheck(self, annotation):
         return
 
 
-class Singleton(type):
+class Intern(type):
```

Comment:
@tpoterba Can you sanity check this?  I don't fully understand this meta class business, and I think you wrote this originally.

Reply:
yeah, this seems fine.